DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/ipv4/test_ipv4_mtu.py

```diff
@@ -0,0 +1,212 @@
+    return stats
+
+
+@pytest.mark.usefixtures("change_port_mtu")
+async def test_ipv4_oversized_mtu(testbed):
+    """
+    Test Name: test_ipv4_oversized_mtu
+    Test Suite: suite_functional_ipv4
+    Test Overview: Test IPv4 oversized mtu counters
+    Test Procedure:
+    1. Init interfaces
+    2. Configure ports up
+    3. Configure IP addrs
+    4. Configure interfaces MTU to 1000
+    5. Generate traffic with packet size 1200 and verify there's no reception due to MTU
+    6. Verify oversized counter been incremented in port statistics
+    """
+    # 1. Init interfaces
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    dent = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[dent][0]
+    ports = tgen_dev.links_dict[dent][1]
+    traffic_duration = 10
+    address_map = (
+        # swp port, tg port,    swp ip,    tg ip,     plen
+        (ports[0], tg_ports[0], "1.1.1.1", "1.1.1.2", 24),
+        (ports[1], tg_ports[1], "2.2.2.1", "2.2.2.2", 24),
+        (ports[2], tg_ports[2], "3.3.3.1", "3.3.3.2", 24),
+        (ports[3], tg_ports[3], "4.4.4.1", "4.4.4.2", 24),
+    )
+    tg_to_swp_map = {
+        tg: swp for swp, tg, *_ in address_map
+    }
+
+    # 2. Configure ports up
+    out = await IpLink.set(input_data=[{dent: [
+        {"device": port, "operstate": "up"}
+        for port, *_ in address_map
+    ]}])
+    assert out[0][dent]["rc"] == 0, "Failed to set port state UP"
+
+    # 3. Configure IP addrs
+    out = await IpAddress.add(input_data=[{dent: [
+        {"dev": port, "prefix": f"{ip}/{plen}"}
+        for port, _, ip, _, plen in address_map
+    ]}])
+    assert out[0][dent]["rc"] == 0, "Failed to add IP addr to port"
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {"ixp": port, "ip": ip, "gw": gw, "plen": plen}
+        for _, port, gw, ip, plen in address_map
+    )
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    streams = {
+        f"{tg1} <-> {tg2}": {
+            "type": "ipv4",
+            "ip_source": dev_groups[tg1][0]["name"],
+            "ip_destination": dev_groups[tg2][0]["name"],
+            "protocol": "ip",
+            "rate": "1000",  # pps
+            "frameSize": 1200,
+        } for tg1, tg2 in zip(tg_ports, reversed(tg_ports))
+    }
+
+    await tgen_utils_setup_streams(tgen_dev, None, streams)
+
+    # 4. Port mtu should be configured in the change_port_mtu fixture
+
+    # 5. Generate traffic with packet size 1200
+    old_stats = await get_port_stats(dent, (port for port, *_ in address_map))
+
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(traffic_duration)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    await asyncio.sleep(traffic_duration)  # wait for delayed stats to update
```

Comment:
It's not clear why you wait traffic_duration. It's better to have another variable, even with the same value.